src/modules/core/components/Fields/Select/Select.tsx

```diff
@@ -255,6 +255,8 @@ const Select = ({
           activeOption.label,
           activeOption.labelValues,
         );
+      } else if (activeOption.labelElement) {
+        activeOptionLabel = activeOption.labelElement;
       } else {
         activeOptionLabel = activeOption.label;
       }
```

Comment:
I mean, I would have just extended `activeOption.label` to be able to support react components as well as strings / message descriptors.

But it's your decision ultimately how you want to implement this.

Reply:
@rdig, I actually did go this route initially. But the way label is used in `SelectOption` & also because of the types, there would need to be more of redoing Select & supporting components and types. So I changed to the `labelElement` version.